source/lib/thunderbird.js

```diff
@@ -64,6 +74,71 @@ function showNewMessageNotification(message, isRSS) {
         text: text,
         iconURL: utils.getIcon(),
     });
+
+}
+
+function format(message, format, callback){
+
+  var author = gHeaderParser.parseDecodedHeader(message.mime2DecodedAuthor);
+
```

Comment:
This might look weird, but obtaining the body of a message is asynchronous, which makes me think that it's not immediate, so I only use it if necessary. This is the reason why `format` is also asynchronous.